### PR TITLE
IGVF-1881-file-set-type-defs

### PR DESCRIPTION
**Pull request by ian-whaling**

### CODE REVIEW

File: src/igvfd/schemas/model_set.json [hunk 1]
```diff
@@ -96,12 +96,19 @@
             "description": "The category that best describes this predictive model set.",
             "comment": "File set type describes the framework of the model in this model set.",
             "enum": [
+                "decision tree",
                 "neural network",
                 "random forest",
-                "decision tree",
                 "support vector machine",
                 "variant binding effect"
```
Comment (ian-whaling) [inline src/igvfd/schemas/model_set.json:103]: One of these is not like the others

File: src/igvfd/schemas/analysis_set.json [hunk 2]
```diff
@@ -71,12 +71,15 @@
...
         "file_set_type": {
-            "description": "The level of this analysis set. An intermediate analysis cannot be interpreted on its own and is part of a principal analysis. A principal analysis is the core analysis for an experimental design, the results of which can be interpretable on their own.",
-            "comment": "File set type describes the content of the analysis set. Intermediate analyses are processed data which are not the final results of an experiment. Principal analyses are processed data which are the final results of an experiment.",
+            "description": "The level of this analysis set.",
             "enum": [
                 "intermediate analysis",
                 "principal analysis"
             ],
+            "enum_descriptions": {
+                "intermediate analysis": "An intermediate analysis is processed data that is not the final result of an experiment and cannot be interpreted on its own, forming part of a principal analysis.",
```
Comment (hitz) [inline src/igvfd/schemas/analysis_set.json:80]: replace "and cannot be interpreted on its own, forming part of a principal analysis"  with ", typically used as input to an principle analysis"

File: src/igvfd/schemas/analysis_set.json [hunk 3]
```diff
@@ -71,12 +71,15 @@
...
+                "principal analysis": "The final, processed data of an experiment, serving as the core analysis, with results that are interpretable on their own."
```
Comment (hitz) [inline src/igvfd/schemas/analysis_set.json:81]: cut "with results that are interpretable on their own"

File: src/igvfd/schemas/auxiliary_set.json [hunk 4]
```diff
@@ -65,6 +65,16 @@
                 "quantification DNA barcode sequencing",
                 "variant sequencing"
             ],
+            "enum_descriptions": {
+                "cell hashing barcode sequencing": "Sequencing of the antibody barcoding of cell batches.",
+                "cell sorting": "Data collected during fluorescence activated cell sorting.",
+                "circularized RNA barcode detection": "Reporter detection (such as for MPRA) through sequencing circularized barcoded RNA molecules.",
+                "gRNA sequencing": "Sequencing data of guide RNAs for CRISPR screen assays.",
```
Comment (hitz) [inline src/igvfd/schemas/auxiliary_set.json:72]: drop word "screen", (its 90% true but CRISPR is used for things other than screening)

File: src/igvfd/schemas/auxiliary_set.json [hunk 5]
```diff
@@ -65,6 +65,16 @@
                 "quantification DNA barcode sequencing",
                 "variant sequencing"
             ],
+            "enum_descriptions": {
```
Comment (hitz) [inline src/igvfd/schemas/auxiliary_set.json:68]: I think these would be clearly if you indicate WHY cell hashing, lipid conjugation, etc is used (I think to extend the barcoding space, but not 100%)

File: src/igvfd/schemas/auxiliary_set.json [hunk 6]
```diff
@@ -65,6 +65,16 @@
...
+                "cell hashing barcode sequencing": "Sequencing of the antibody barcoding of cell batches.",
+                "cell sorting": "Data collected during fluorescence activated cell sorting.",
+                "circularized RNA barcode detection": "Reporter detection (such as for MPRA) through sequencing circularized barcoded RNA molecules.",
+                "gRNA sequencing": "Sequencing data of guide RNAs for CRISPR screen assays.",
+                "lipid-conjugated oligo sequencing": "Lipid-conjugated oligo sequencing is lipid- and cholesterol-modified oligonucleotides.",
+                "MORF barcode sequencing": "MORF barcode sequencing is the sequencing of the Multiplexed Overexpression of Regulatory Factors library, a collection of human transcription factor open reading frames, to produce a mapping of barcodes to an overexpressed transcription factor.",
+                "quantification DNA barcode sequencing": "Sequencing data of DNA barcodes for MPRAs.",
```
Comment (hitz) [inline src/igvfd/schemas/auxiliary_set.json:75]: replace MRPAs with "reporter assays"

File: src/igvfd/schemas/construct_library_set.json [hunk 7]
```diff
@@ -694,13 +694,19 @@
...
-            "comment": "File set type describes the content of the construct library set. Guide libraries are guides used for CRISPR screens. Reporter libraries are reporters used for MPRA. Expression vector libraries are used when the content of the vector (a specific allele, gene, exon, or tile of a protein) will be expressed in the target cells. Editing template libraries are used for SGE.",
+            "comment": "File set type describes the content of the construct library set.",
             "enum": [
                 "guide library",
                 "reporter library",
                 "expression vector library",
                 "editing template library"
             ],
+            "enum_descriptions": {
+                "guide library": "Guide libraries are guides used for CRISPR screens.",
```
Comment (hitz) [inline src/igvfd/schemas/construct_library_set.json:705]: The list of guide RNAs used in CRISPR experiments

File: src/igvfd/schemas/construct_library_set.json [hunk 8]
```diff
@@ -694,13 +694,19 @@
...
+                "reporter library": "Reporter libraries are reporters used for MPRA.",
```
Comment (hitz) [inline src/igvfd/schemas/construct_library_set.json:706]: the library of elements used in a massively parallel reporter assay (MPRA)

File: src/igvfd/schemas/curated_set.json [hunk 9]
```diff
@@ -69,6 +69,17 @@
                 "transcriptome",
                 "variants"
             ],
+            "enum_descriptions": {
+                "barcodes": "Barcodes are lists of barcodes.",
```
Comment (hitz) [inline src/igvfd/schemas/curated_set.json:73]: ... lists of barcodes identifying cells in a single cell experiment.

File: src/igvfd/schemas/curated_set.json [hunk 10]
```diff
@@ -69,6 +69,17 @@
...
+                "editing templates": "Editing templates are used in Saturation Genome Editing (SGE) assays.",
+                "elements": "Elements are lists of elements in the genome.",
```
Comment (hitz) [inline src/igvfd/schemas/curated_set.json:75]: ...lists of genomic elements to be assayed

File: src/igvfd/schemas/curated_set.json [hunk 11]
```diff
@@ -69,6 +69,17 @@
...
+                "external data for catalog": "External data for catalog are reference files from other sources utilized in the IGVF catalog.",
```
Comment (hitz) [inline src/igvfd/schemas/curated_set.json:76]: public reference files

File: src/igvfd/schemas/curated_set.json [hunk 12]
```diff
@@ -69,6 +69,17 @@
...
+                "genome": "Genome file sets are genome references.",
```
Comment (hitz) [inline src/igvfd/schemas/curated_set.json:77]: Genome file sets contain files representing a reference genome

**Review by hitz — CHANGES_REQUESTED** [review]

File: src/igvfd/schemas/analysis_set.json (hunk 2 above)
Comment (ian-whaling) [reply src/igvfd/schemas/analysis_set.json:80]: fixed

File: src/igvfd/schemas/analysis_set.json (hunk 3 above)
Comment (ian-whaling) [reply src/igvfd/schemas/analysis_set.json:81]: fixed

File: src/igvfd/schemas/auxiliary_set.json (hunk 4 above)
Comment (ian-whaling) [reply src/igvfd/schemas/auxiliary_set.json:72]: fixed

File: src/igvfd/schemas/auxiliary_set.json (hunk 6 above)
Comment (ian-whaling) [reply src/igvfd/schemas/auxiliary_set.json:75]: fixed

File: src/igvfd/schemas/construct_library_set.json (hunk 8 above)
Comment (ian-whaling) [reply src/igvfd/schemas/construct_library_set.json:706]: fixed

File: src/igvfd/schemas/construct_library_set.json (hunk 7 above)
Comment (ian-whaling) [reply src/igvfd/schemas/construct_library_set.json:705]: fixed

File: src/igvfd/schemas/curated_set.json (hunk 9 above)
Comment (ian-whaling) [reply src/igvfd/schemas/curated_set.json:73]: fixed

File: src/igvfd/schemas/curated_set.json (hunk 10 above)
Comment (ian-whaling) [reply src/igvfd/schemas/curated_set.json:75]: fixed

File: src/igvfd/schemas/curated_set.json (hunk 11 above)
Comment (ian-whaling) [reply src/igvfd/schemas/curated_set.json:76]: fixed

File: src/igvfd/schemas/curated_set.json (hunk 12 above)
Comment (ian-whaling) [reply src/igvfd/schemas/curated_set.json:77]: fixed

File: src/igvfd/schemas/auxiliary_set.json (hunk 5 above)
Comment (ian-whaling) [reply src/igvfd/schemas/auxiliary_set.json:68]: added purposes

**Review by hitz — APPROVED** [review]

**Pull request merged**